File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1837_basico.js
import { question } from "readline-sync";

function main() {
    let entrada = get_Text().split(` `);
    let denominador = Number(entrada[0]);
    let divisor = Number(entrada[1]);

    let quociente = 0;
    let resto = 0;
    if(divisor < 0) {
        divisor = divisor * (-1);
        quociente = Math.floor((denominador) / (divisor));
        quociente = quociente * (-1);
        divisor = divisor * (-1);
        resto = (denominador) - ((divisor) * quociente);

    }else {
        quociente = Math.floor((denominador) / (divisor));
        resto = (denominador) - ((divisor) * quociente);
    }

    print(`${quociente} ${resto}`);

}

function get_Text() {
    return question();
}

function print(texto) {
    return console.log(texto);
}

main();

File: Exercicios_ED_I/Bee/1837_basico.cpp
#include <stdio.h>
#include <iostream>
#include <math.h>
using namespace std;

main() {
    int a, b;
    cin >> a >> b;

    int resto, quociente;

    if (b < 0) {
        b = b * -1;
        quociente = floor(a / static_cast<double>(b));
        quociente = quociente * -1;
        b = b * -1;
        resto = a - ((b) * quociente); 
    } else {
        quociente = floor(a / static_cast<double>(b));
        resto = a - ((b) * quociente); 
    }

    cout << quociente << " " << resto << endl;
}